app/controllers/api/v1/orders_packages_controller.rb

```diff
@@ -28,6 +28,7 @@ def index
         return search_by_package_id if params['search_by_package_id'].present?
         # needs to be removed as it makes unwanted orders_packages request and makes the app slow
         return all_orders_packages if params['all_orders_packages'].present?
+        return orders_package_by_order_id if params['order_id'].present?
```

Comment:
Note: all of the query parameters of this endpoint are _exclusive_ of each other, which is generally not how we want to build an entry point. Rather we want to be able to use the params in a _complementary_ way to allow a lot of flexibility in its usage.

No need to change anything, as it would mean a lot of extra work.
But to keep in mind for the future, if there's any questions I'm always happy to help.

cc - @abhinav098 @namrataukirde @abhinav098 @akshaygwd @steveyken